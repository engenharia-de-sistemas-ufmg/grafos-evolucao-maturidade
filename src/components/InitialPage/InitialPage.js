import React from 'react';
import PropTypes from 'prop-types';

import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';

import * as labels from '../../commons/labels';

const InitialPage = ({ language, loadPage, updateGraph }) => {
  const handleBasicGraph = () => {
    updateGraph(basicGraph);
    loadPage();
  };

  const handleSystemEngineer = () => {
    updateGraph(systemEngineerGraph);
    loadPage();
  };

  return (
    <div>
      <div>{labels.initialPage[language]}</div>
      <button type="button" className="btn-basic-graph" onClick={handleBasicGraph}>
        {labels.basicModel[language]}
      </button>
      <button type="button" className="btn-system-engineer" onClick={handleSystemEngineer}>
        {labels.systemEngineer[language]}
      </button>
    </div>
  );
};

InitialPage.propTypes = {
  language: PropTypes.string.isRequired,
  loadPage: PropTypes.func.isRequired,
  updateGraph: PropTypes.func.isRequired,
};

export default InitialPage;
