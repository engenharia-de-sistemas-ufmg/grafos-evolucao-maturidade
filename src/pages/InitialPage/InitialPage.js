import React from 'react';
import PropTypes from 'prop-types';

import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';

import * as labels from '../../commons/labels';

const InitialPage = ({ lang, viewGraph, updateGraph }) => {
  const handleBasicGraph = () => {
    updateGraph(basicGraph);
    viewGraph();
  };

  const handleSystemEngineer = () => {
    updateGraph(systemEngineerGraph);
    viewGraph();
  };

  const title = (
    <header className="bd-header">
      <h1 className="title">
        {labels.maturityEvolutionGraphsTitle[lang]}
      </h1>
      <h2 className="subtitle">
        {labels.maturityEvolutionGraphsSubtitle[lang]}
      </h2>
      <p>&nbsp;</p>
    </header>
  );

  const button = (label, subtitle, buttonClass, click) => (
    <div className="box">
      <article className="media">
        <div className="media-left">
          <button type="button" className={`${buttonClass} button is-primary`} onClick={click}>
            {label}
          </button>
        </div>
        <div className="media-content">
          {subtitle}
        </div>
      </article>
    </div>
  );

  const buttonBasicGraph = button(labels.basicModel[lang],
    labels.basicModelSubtitle[lang],
    'btn-basic-graph',
    handleBasicGraph);

  const buttonSystemEngineer = button(labels.systemEngineer[lang],
    labels.systemEngineerSubtitle[lang],
    'btn-system-engineer',
    handleSystemEngineer);

  return (
    <section className="section">
      <div className="container">
        <div className="notification">
          {title}
          {buttonBasicGraph}
          {buttonSystemEngineer}
        </div>
      </div>
    </section>
  );
};

InitialPage.propTypes = {
  lang: PropTypes.string.isRequired,
  viewGraph: PropTypes.func.isRequired,
  updateGraph: PropTypes.func.isRequired,
};

export default InitialPage;
