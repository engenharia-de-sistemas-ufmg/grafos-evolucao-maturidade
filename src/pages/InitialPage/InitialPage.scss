.is-right {
  float: right;
}

.graph-example-container {
  display: flex;
  margin-bottom: 12px;
}

.graph-example-container > button {
  display: block;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 0;
  padding-right: 8px;
  width: 25%;
  height: auto;
  text-align: center;
  white-space: normal;
}

.graph-example-container > p {
  display: block;
  flex-basis: auto;
  flex-grow: 1;
  flex-shrink: 1;
  padding-left: 8px;
  width: 75%;
}
