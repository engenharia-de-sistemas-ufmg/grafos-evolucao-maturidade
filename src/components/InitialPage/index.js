import { connect } from 'react-redux';
import InitialPage from './InitialPage';
import { updateGraph, loadPage } from '../../actions';

const mapDispatchToProps = (dispatch) => ({
  loadPage: () => { dispatch(loadPage()); },
  updateGraph: (graph) => { dispatch(updateGraph(graph)); },
});

export default connect(
  null,
  mapDispatchToProps,
)(InitialPage);
