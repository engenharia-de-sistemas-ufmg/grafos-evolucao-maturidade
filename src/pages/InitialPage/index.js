import { connect } from 'react-redux';
import InitialPage from './InitialPage';
import { updateGraph } from '../../actions/graph';
import { viewGraph } from '../../actions/page';

const mapStateToProps = (state) => ({
  lang: state.page.lang,
});

const mapDispatchToProps = (dispatch) => ({
  viewGraph: () => { dispatch(viewGraph()); },
  updateGraph: (graph) => { dispatch(updateGraph(graph)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(InitialPage);
