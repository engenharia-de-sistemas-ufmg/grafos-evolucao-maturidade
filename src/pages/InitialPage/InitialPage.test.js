import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import InitialPage from './InitialPage';
import basicGraph from '../../examples/basicGraph';
import systemEngineerGraph from '../../examples/systemEngineerGraph';

describe('InitialPage', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const lang = 'pt';
    const updateNodeGroups = () => { };
    const updateGraph = () => { };
    const viewGraph = () => { };
    const component = shallow(
      <InitialPage
        lang={lang}
        updateNodeGroups={updateNodeGroups}
        updateGraph={updateGraph}
        viewGraph={viewGraph}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should trigger basicGraph', () => {
    const lang = 'pt';
    const updateNodeGroups = sinon.spy();
    const updateGraph = sinon.spy();
    const viewGraph = sinon.spy();
    const component = shallow(
      <InitialPage
        lang={lang}
        updateNodeGroups={updateNodeGroups}
        updateGraph={updateGraph}
        viewGraph={viewGraph}
      />,
    );
    component.find('.btn-basic-graph').simulate('click');
    expect(viewGraph.callCount).toBe(1);
    expect(updateGraph.callCount).toBe(1);
    expect(updateGraph.args[0][0]).toStrictEqual({
      nodes: basicGraph.nodes,
      edges: basicGraph.edges,
    });
    expect(updateNodeGroups.callCount).toBe(1);
    expect(updateNodeGroups.args[0][0]).toBe(basicGraph.groups);
  });

  test('should trigger systemEngineer', () => {
    const lang = 'pt';
    const updateNodeGroups = sinon.spy();
    const updateGraph = sinon.spy();
    const viewGraph = sinon.spy();
    const component = shallow(
      <InitialPage
        lang={lang}
        updateNodeGroups={updateNodeGroups}
        updateGraph={updateGraph}
        viewGraph={viewGraph}
      />,
    );
    component.find('.btn-system-engineer').simulate('click');
    expect(viewGraph.callCount).toBe(1);
    expect(updateGraph.callCount).toBe(1);
    expect(updateGraph.args[0][0]).toStrictEqual({
      nodes: systemEngineerGraph.nodes,
      edges: systemEngineerGraph.edges,
    });
    expect(updateNodeGroups.callCount).toBe(1);
    expect(updateNodeGroups.args[0][0]).toBe(systemEngineerGraph.groups);
  });
});
