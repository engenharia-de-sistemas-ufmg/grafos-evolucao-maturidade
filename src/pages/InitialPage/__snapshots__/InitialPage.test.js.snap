// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialPage should match snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <div
      className="notification"
    >
      <header
        className="bd-header"
      >
        <h1
          className="title"
        >
          Grafos de Evolução da Maturidade
        </h1>
        <h2
          className="subtitle"
        >
          Uma proposta de representação da evolução do conhecimento por meio de grafos.
        </h2>
        <p>
           
        </p>
      </header>
      <div
        className="box"
      >
        <article
          className="media"
        >
          <div
            className="media-left"
          >
            <button
              className="btn-basic-graph button is-primary"
              onClick={[Function]}
              type="button"
            >
              Modelo básico
            </button>
          </div>
          <div
            className="media-content"
          >
            Um pequeno grafo apenas como exemplo.
          </div>
        </article>
      </div>
      <div
        className="box"
      >
        <article
          className="media"
        >
          <div
            className="media-left"
          >
            <button
              className="btn-system-engineer button is-primary"
              onClick={[Function]}
              type="button"
            >
              Engenharia de Sistemas
            </button>
          </div>
          <div
            className="media-content"
          >
            Grade de disciplinas do curso de graduação em Engenharia de Sistemas da UFMG.
          </div>
        </article>
      </div>
    </div>
  </div>
</section>
`;
